### PR TITLE
Migrate to `fusion core@0.3.0-5`

### LINKED ISSUE #43: Migrate fusion-plugin-rpc to createPlugin

### PR DESCRIPTION
**Pull request by AlexMSmithCA**
Fixes #43 

Includes:
* Update all dependencies to latest
* Migration to `createPlugin` API
* Unit test changes, including new tests to ensure `service` and `middleware` are resolved correctly.
* _(minor)_ Updates to `README.md`

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,18 +36,19 @@ const handlers = __NODE__ && {
...
-  app.configure(FetchToken, fetch);
-
-  app.register(withDependencies({
-    RPC: RPCToken
-  })(({RPC}) => (ctx, next) => {
-    RPC(ctx).request('getUser', 1).then(console.log) // {some: 'data1'}
+  app.register(RPCHandlersToken, handlers);
+  app.register(FetchToken, fetch);
+
+  app.register(createPlugin({
```
Comment (ganemone) [inline README.md]: I think this example would be simpler if you use the `app.middleware` api

File: src/__tests__/index.browser.js [hunk 2]
```diff
@@ -5,45 +5,94 @@
...
+  return app;
+}
+
+test('success status request', t => {
+  const app = createTestFixture();
+
+  let wasResolved = false;
+  getSimulator(
+    app,
+    createPlugin({
```
Comment (ganemone) [inline src/__tests__/index.browser.js]: I didn't realize we supported passing plugins to `getSimulator`. What is the purpose of this vs just calling `app.register` ?

File: src/browser.js [hunk 3]
```diff
@@ -6,50 +6,63 @@
...
 
-import {FetchToken, createToken} from 'fusion-tokens';
-import {withDependencies} from 'fusion-core';
+import {FetchToken, createOptionalToken} from 'fusion-tokens';
+import {createPlugin} from 'fusion-core';
 import {RPCHandlersToken} from './tokens';
 
-export const RPCRoutePrefixConfigToken = createToken(
-  'RPCRoutePrefixConfigToken'
+export const RPCRoutePrefixConfigToken = createOptionalToken(
```
Comment (ganemone) [inline src/browser.js]: I'm not totally sold on this, but we can go with it for now. I think long term we should look into composition (see https://github.com/fusionjs/fusion-core/issues/90)

File: src/__tests__/index.browser.js (hunk 2 above)
Comment (AlexMSmithCA) [reply src/__tests__/index.browser.js]: No tangible difference.  It's just to explicitly denote that the "plugin" there is for testing only.

File: README.md (hunk 1 above)
Comment (AlexMSmithCA) [reply README.md]: Good call, I agree.  I'll update this.

File: src/browser.js (hunk 3 above)
Comment (AlexMSmithCA) [reply src/browser.js]: Agreed this isn't ideal long-term.  I could see us going either the composition route, and would have to provide extensions for all plugins that require `routePrefix` _or_ we move this token into `fusion-tokens` as many plugins will likely come to rely upon this config string.

**Comment by AlexMSmithCA** [comment]
!merge

**Pull request merged**